src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,10 +1753,13 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
-                return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
-            } else {
+                return result;
```

Comment:
Oh, right, and we don't throw this error because we will try second attempt to get token...